File: Modules/HumanResource/PayrollSettings/EmployerContributions/EmployerContributionsRow.cs
using HRMSoftware.Administration;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace HRMSoftware.PayrollSettings;

[ConnectionKey("Default"), Module("PayrollSettings"), TableName("HumanResourcesPayrollEmployerContributions")]
[DisplayName("Employer Contributions"), InstanceName("Employer Contributions")]
[ReadPermission(PermissionKeys.HumanResources)]
[ModifyPermission(PermissionKeys.HumanResources)]
public sealed class EmployerContributionsRow : LoggingRow<EmployerContributionsRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Column("ID"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Payslip Id")]
    public int? PayslipId
    {
        get => fields.PayslipId[this];
        set => fields.PayslipId[this] = value;
    }

    [DisplayName("Employer Epf"), Column("Amount")]
    public double? Amount
    {
        get => fields.Amount[this];
        set => fields.Amount[this] = value;
    }
    [DisplayName("Description"), NameProperty]
    public string Description
    {
        get => fields.Description[this];
        set => fields.Description[this] = value;
    }
    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public Int32Field PayslipId;
        public DoubleField Amount;
        public StringField Description;

    }
}

File: Modules/ServerTypes/PayrollSettings/EmployerContributionsRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface EmployerContributionsRow {
    Id?: number;
    PayslipId?: number;
    Amount?: number;
    Description?: string;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
    DeleteUserId?: number;
    DeleteDate?: string;
    IsActive?: number;
}

export abstract class EmployerContributionsRow {
    static readonly idProperty = 'Id';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'Description';
    static readonly localTextPrefix = 'PayrollSettings.EmployerContributions';
    static readonly deletePermission = 'Administration:HumanResources';
    static readonly insertPermission = 'Administration:HumanResources';
    static readonly readPermission = 'Administration:HumanResources';
    static readonly updatePermission = 'Administration:HumanResources';

    static readonly Fields = fieldsProxy<EmployerContributionsRow>();
}